### PR TITLE
feat: Create standalone components required for mobile search results page.

### PR DESCRIPTION
**Pull request by deepakvenkat**
## What
Creates the following components: 
1. ChipList - A list of chips with additional logic to display only 2 rows of chips and show a `+ more` chip
2. StarRating - A component to show and interact with star ratings for a business
3. FilterDialog - A component to show the filters and manage the current value of filters. 

## Ticket
https://app.clubhouse.io/out-in-tech/story/85/fe-scaffolding-space-results-mobile-page

### CODE REVIEW

File: src/components/FilterDialog/FilterDialog.jsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+          className={classes.closeButton}
+        >
+          <CloseIcon />
+        </IconButton>
+      </Toolbar>
+      <div className={classes.dialogBody}>
+        <Typography variant="h5">Filter by</Typography>
+        <div className={classes.section}>
+          <Typography variant="h6">Average Rating</Typography>
+          <StarRating numberFilled={stars} onRatingChanged={(value) => dispatch({ type: 'start', payload: { stars: value } })} />
```
Comment (deepakvenkat) [inline src/components/FilterDialog/FilterDialog.jsx:86]: Fix typo `star`

File: package.json [hunk 2]
```diff
@@ -8,9 +8,11 @@
     "@testing-library/jest-dom": "^5.11.4",
     "@testing-library/react": "^11.1.0",
     "@testing-library/user-event": "^12.1.10",
+    "classnames": "^2.2.6",
     "prop-types": "^15.7.2",
     "react": "^17.0.1",
     "react-dom": "^17.0.1",
+    "react-intersection-observer": "^8.31.0",
```
Comment (deepakvenkat) [inline package.json:15]: I am using this in `ChipList` to detect whether a chip is in view or not

File: src/components/ChipList/ChipList.jsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    maxHeight: 80,
+    overflow: 'hidden',
+  },
+  chip: {
+    marginBottom: 8,
+    marginRight: 4,
+  },
+};
+
+const ChipList = ({ chips, classes }) => {
```
Comment (deepakvenkat) [inline src/components/ChipList/ChipList.jsx:18]: The logic here is : 
1. Display only two lines chips by setting a maxHeight on the parent. 
2. Hide the other chips using `overflow: hidden` on the parent. 
3. Add an intersectionobserver to each chip
4. After rendering in `useLayoutEffect` figure out how many are not in view and set that in state.
5. Use that state variable to show `+ count more` 

I couldn't think of another way of doing this but open to suggestions.

File: package.json [hunk 4]
```diff
@@ -8,9 +8,11 @@
     "@testing-library/jest-dom": "^5.11.4",
     "@testing-library/react": "^11.1.0",
     "@testing-library/user-event": "^12.1.10",
+    "classnames": "^2.2.6",
```
Comment (deepakvenkat) [inline package.json:11]: I am not using this yet, but figured might need it sooner or later so left it in here.

File: package.json (hunk 2 above)
Comment (InManuBytes) [reply package.json:15]: Haven't used it, but looks well-maintained.

File: src/components/BusinessCard/BusinessCard.jsx [hunk 5]
```diff
@@ -70,79 +71,55 @@ const BusinessCard = ({
...
-            ''
-          )}
+      )}
+      action={<BookmarkBorderIcon fontSize="large" color="secondary" />}
+      title={<Typography variant="h6">{name}</Typography>}
+      subheader={<Typography variant="body2">{category}</Typography>}
+    />
+    <CardMedia
+      image={imageUrl}
+      // todo make this responsive
```
Comment (InManuBytes) [inline src/components/BusinessCard/BusinessCard.jsx:95]: I'm guessing like this: https://material-ui.com/components/cards/#RecipeReviewCard.js
It's not part of this ticket though is it?

File: src/components/ChipList/ChipList.jsx (hunk 3 above)
Comment (InManuBytes) [reply src/components/ChipList/ChipList.jsx:18]: Not sure if you saw this section in material UI: https://material-ui.com/components/hidden/
Specifically like this: https://material-ui.com/components/hidden/#BreakpointDown.js

**Review by InManuBytes — APPROVED** [review]
Aside from the chips breakpoint, everything else LGTM. I don't want that to be a blocker though if it works.

File: src/components/ChipList/ChipList.jsx (hunk 3 above)
Comment (deepakvenkat) [reply src/components/ChipList/ChipList.jsx:18]: I did not see that. I will see if I can change it in a followup pr.

**Pull request merged**